src/zenml/model/model_config.py

```diff
@@ -57,15 +228,17 @@ def get_or_create_model(self) -> "ModelResponseModel":
         Returns:
             The model based on configuration.
         """
-        if self._model is not None:
-            return self._model
+        from zenml.models.model_models import ModelResponseModel
+
+        if self.model is not None:
+            return ModelResponseModel(**dict(self.model))
```

Comment:
Same here, isn't `self.model` already a `ModelResponseModel`?

Reply:
I'll redesign this

Reply:
I decided to get rid of this caching, as it produces unexpected results and always returns the version cached at the very beginning of the pipeline run.